manifests/stage-ib-kubernetes/0040-deployment.yaml

```diff
@@ -0,0 +1,124 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+  name: ib-kubernetes
+  namespace: {{ .RuntimeSpec.Namespace }}
+  annotations:
+    kubernetes.io/description: |
+      This deployment launches the ib-kubernetes daemon for Infiniband CNI.
+spec:
+  progressDeadlineSeconds: 600
+  strategy:
+    type: RollingUpdate
+  replicas: 1
+  selector:
+    matchLabels:
+      name: ib-kubernetes
+  template:
+    metadata:
+      labels:
+        name: ib-kubernetes
+        component: network
+        type: infra
+        kubernetes.io/os: "linux"
+    spec:
+      priorityClassName: system-cluster-critical
+      serviceAccountName: ib-kubernetes
+      hostNetwork: true
+      # required to be scheduled on a linux node with node-role.kubernetes.io/master label and
+      # only one instance of ib-kubernetes pod per node
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+              - matchExpressions:
+                  - key: node-role.kubernetes.io/master
+                    operator: In
+                    values:
+                      - ""
+                  - key: kubernetes.io/os
+                    operator: In
+                    values:
+                      - "linux"
+              - matchExpressions:
+                  - key: node-role.kubernetes.io/control-plane
+                    operator: In
+                    values:
+                      - ""
+                  - key: kubernetes.io/os
+                    operator: In
+                    values:
+                      - "linux"
+        podAntiAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            - labelSelector:
+                matchExpressions:
+                  - key: name
+                    operator: In
+                    values:
+                      - ib-kubernetes
+              topologyKey: kubernetes.io/hostname
+      containers:
+        - name: ib-kubernetes
+          image: {{ .CrSpec.Repository }}/{{ .CrSpec.Image }}:{{ .CrSpec.Version }}
+          imagePullPolicy: IfNotPresent
+          command: ["/usr/bin/ib-kubernetes"]
+          resources:
+            requests:
+              cpu: 100m
+              memory: 300Mi
+          env:
+            - name: DAEMON_SM_PLUGIN
+              value: "ufm"
+            - name: DAEMON_PERIODIC_UPDATE
+              value: "5"
+            - name: GUID_POOL_RANGE_START
+              value: {{ .CrSpec.PKeyGUIDPoolRangeStart }}
+            - name: GUID_POOL_RANGE_END
+              value: {{ .CrSpec.PKeyGUIDPoolRangeEnd }}
+            - name: UFM_USERNAME
+              valueFrom:
+                secretKeyRef:
+                  name: {{ .CrSpec.UfmSecret }}
+                  key: UFM_USERNAME
+                  optional: true
```

Comment:
According to this [1] code, username, password and address are mandatory fields

[1] https://github.com/Mellanox/ib-kubernetes/blob/803881ec421404170181a39f7ec255f82db8645f/pkg/sm/plugins/ufm/ufm.go#L45